api/src/main/java/org/openmrs/module/fhir/api/util/BaseOpenMRSDataUtil.java

```diff
@@ -0,0 +1,250 @@
+package org.openmrs.module.fhir.api.util;
+
+import org.hl7.fhir.dstu3.model.BooleanType;
+import org.hl7.fhir.dstu3.model.DateTimeType;
+import org.hl7.fhir.dstu3.model.DomainResource;
+import org.hl7.fhir.dstu3.model.Extension;
+import org.hl7.fhir.dstu3.model.PrimitiveType;
+import org.hl7.fhir.dstu3.model.StringType;
+import org.openmrs.BaseOpenmrsData;
+import org.openmrs.BaseOpenmrsMetadata;
+import org.openmrs.User;
+import org.openmrs.api.context.Context;
+
+import java.util.Date;
+
+public final class BaseOpenMRSDataUtil {
+
+    /** https://simplifier.net/SRxProject/resource-date-created/~json */
+    private static final String DATE_CREATED_URL = "http://fhir-es.transcendinsights.com/stu3/StructureDefinition/resource-date-created";
+
+    /** https://simplifier.net/eLabTest/Creator-crew-version1/~json */
+    private static final String CREATOR_URL = "https://purl.org/elab/fhir/StructureDefinition/Creator-crew-version1";
+
+    // Local extensions
+    private static final String CHANGED_BY_URL = "changedBy";
+    private static final String DATE_CHANGED_URL = "dateChanged";
+    private static final String VOIDED_URL = "voided";
+    private static final String DATE_VOIDED_URL = "dateVoided";
+    private static final String VOIDED_BY_URL = "voidedBy";
+    private static final String VOID_REASON_URL = "voidReason";
+    private static final String RETIRED_URL = "retired";
+    private static final String DATE_RETIRED_URL = "dateRetired";
+    private static final String RETIRED_BY_URL = "retiredBy";
+    private static final String RETIRE_REASON_URL = "retireReason";
+
+    private BaseOpenMRSDataUtil() { }
+
+    public static void setBaseExtensionFields(DomainResource fhirResource, BaseOpenmrsData openmrsData) {
+        fhirResource.addExtension(createDateCreatedExtension(openmrsData.getDateCreated()));
+        fhirResource.addExtension(createCreatorExtension(openmrsData.getCreator()));
+
+        if (openmrsData.getDateChanged() != null) {
+            fhirResource.addExtension(createDateChangedExtension(openmrsData.getDateChanged()));
+            fhirResource.addExtension(createChangedByExtension(openmrsData.getChangedBy()));
+        }
+        if (openmrsData.getVoided()) {
+            fhirResource.addExtension(createVoidedExtension(openmrsData.getVoided()));
+            fhirResource.addExtension(createDateVoidedExtension(openmrsData.getDateVoided()));
+            fhirResource.addExtension(createVoidedByExtension(openmrsData.getVoidedBy()));
+            fhirResource.addExtension(createVoidReasonExtension(openmrsData.getVoidReason()));
+        }
+    }
+    
+    public static void setBaseExtensionFields(DomainResource fhirResource, BaseOpenmrsMetadata openmrsMetadata) {
+        fhirResource.addExtension(createDateCreatedExtension(openmrsMetadata.getDateCreated()));
+        fhirResource.addExtension(createCreatorExtension(openmrsMetadata.getCreator()));
+
+        if (openmrsMetadata.getDateChanged() != null) {
```

Comment:
Shouldn't we check changedBy as well?

Reply:
I assumed that if dateChanged is present then changedBy is also present.

Reply:
It should be true, so maybe your right.